python_src/src/lamp_py/performance_manager/l1_rt_trips.py

```diff
@@ -392,10 +422,17 @@ def update_static_trip_id_guess_exact(db_manager: DatabaseManager) -> None:
                 StaticTrips.static_version_key
                 == TempEventCompare.static_version_key,
                 StaticTrips.trip_id == TempEventCompare.trip_id,
-                StaticTrips.direction_id == TempEventCompare.direction_id,
-                StaticTrips.route_id == TempEventCompare.route_id,
             ),
         )
+        .join(
+            static_stop_sub,
+            sa.and_(
+                static_stop_sub.c.static_version_key
+                == TempEventCompare.static_version_key,
+                static_stop_sub.c.trip_id == TempEventCompare.trip_id,
+            ),
+        )
+        .where(TempEventCompare.start_time.is_not(None))
```

Comment:
this doesn't have any unintended filtering?

it seems like we'll get results in this subquery with it added in?